homeassistant/components/tuya/alarm_control_panel.py

```diff
@@ -0,0 +1,139 @@
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_TRIGGER,
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantTuyaData
+from .base import TuyaEntity
+from .const import DOMAIN, TUYA_DISCOVERY_NEW, DPCode
+
+ICON = "mdi:security"
+
+MODE_ALARM_ARM = "arm"
+MODE_ALARM_DISARMED = "disarmed"
+MODE_ALARM_HOME = "home"
+MODE_ALARM_SOS = "sos"
+
+
+@dataclass
+class TuyaAlarmControlPanelEntityDescription(AlarmControlPanelEntityDescription):
+    """Describe an Tuya Alarm entity."""
+
+
+# All descriptions can be found here:
+# https://developer.tuya.com/en/docs/iot/standarddescription?id=K9i5ql6waswzq
+ALARM: dict[str, tuple[TuyaAlarmControlPanelEntityDescription, ...]] = {
+    # Alarm Host
+    # https://developer.tuya.com/en/docs/iot/categorymal?id=Kaiuz33clqxaf
+    "mal": (
+        TuyaAlarmControlPanelEntityDescription(
+            key=DPCode.MASTER_MODE,
+            name="Alarm",
+        ),
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tuya alarm dynamically through Tuya discovery."""
+    hass_data: HomeAssistantTuyaData = hass.data[DOMAIN][entry.entry_id]
+
+    @callback
+    def async_discover_device(device_ids: list[str]) -> None:
+        """Discover and add a discovered Tuya siren."""
+        entities: list[TuyaAlarmEntity] = []
+        for device_id in device_ids:
+            device = hass_data.device_manager.device_map[device_id]
+            if descriptions := ALARM.get(device.category):
+                for description in descriptions:
+                    if (
+                        description.key in device.function
+                        or description.key in device.status
+                    ):
+                        entities.append(
+                            TuyaAlarmEntity(
+                                device, hass_data.device_manager, description
+                            )
+                        )
+        async_add_entities(entities)
+
+    async_discover_device([*hass_data.device_manager.device_map])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, TUYA_DISCOVERY_NEW, async_discover_device)
+    )
+
+
+class TuyaAlarmEntity(TuyaEntity, AlarmControlPanelEntity):
+    """Tuya Alarm Entity."""
+
+    entity_description: TuyaAlarmControlPanelEntityDescription
+    _attr_supported_features = (
+        SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_TRIGGER
```

Comment:
These capabilities need to be detected and checked if supported by the device. not all devices support away mode